nuxeo-services/nuxeo-platform-dublincore/src/main/java/org/nuxeo/ecm/platform/dublincore/service/DublinCoreStorageService.java

```diff
@@ -76,35 +78,44 @@ public void addContributor(DocumentModel doc, Event event) {
             }
         }
 
-        String[] contributorsArray = (String[]) doc.getProperty("dublincore", "contributors");
-
-        List<String> contributorsList = new ArrayList<String>();
-
-        if (contributorsArray != null && contributorsArray.length > 0) {
-            contributorsList = Arrays.asList(contributorsArray);
-            // make it resizable
-            contributorsList = new ArrayList<String>(contributorsList);
-        } else {
-            // initialize creator too
-            SchemaManager schemaMgr = Framework.getService(SchemaManager.class);
-            if (schemaMgr.getSchema("dublincore").getField("creator") != null) {
-                // First time only => creator
-                doc.setProperty("dublincore", "creator", principalName);
-            }
+        if (doc.getProperty("dublincore", "creator") == null) {
+            // First time only => creator
+            doc.setProperty("dublincore", "creator", principalName);
         }
 
+        List<String> contributorsList = getExistingContributors(doc);
         if (!contributorsList.contains(principalName)) {
             contributorsList.add(principalName);
-            String[] contributorListIn = new String[contributorsList.size()];
-            contributorsList.toArray(contributorListIn);
-            doc.setProperty("dublincore", "contributors", contributorListIn);
         }
-
+        doc.setProperty("dublincore", "contributors", contributorsList.toArray());
```

Comment:
You shouldn't need to call `toArray()`, you can pass a `List` and it will be converted automatically.